### ISSUE TITLE
Error in oldname[j, 1:2] : subscript out of bounds

### ISSUE THREAD

**Issue by INFINITY1993** [DaliangNing/iCAMP1#22]
hello, when i run icamp.bins (# 10 # iCAMP bin level statistics),  i met the following error: Error in oldname[j, 1:2] : subscript out of bounds
Calls: icamp.bins -> <Anonymous> -> t -> sapply -> lapply -> FUN
Execution halted

The nohup.out:
nohup: ignoring input
Now summarizing method=CbMPDiCbraya i=1 j=1. Sun Feb 20 21:25:44 2022
bootstrapping rt=1. Sun Feb 20 21:25:45 2022
bootstrapping rt=201. Sun Feb 20 21:26:04 2022
bootstrapping rt=401. Sun Feb 20 21:26:22 2022
bootstrapping rt=601. Sun Feb 20 21:26:42 2022
bootstrapping rt=801. Sun Feb 20 21:27:00 2022
Now summarizing method=CbMPDiCbraya i=1 j=2. Sun Feb 20 21:27:19 2022
bootstrapping rt=1. Sun Feb 20 21:27:20 2022
bootstrapping rt=201. Sun Feb 20 21:27:38 2022
bootstrapping rt=401. Sun Feb 20 21:27:57 2022
bootstrapping rt=601. Sun Feb 20 21:28:16 2022
bootstrapping rt=801. Sun Feb 20 21:28:35 2022
Now summarizing method=CbMPDiCbraya i=1 j=3. Sun Feb 20 21:28:54 2022
bootstrapping rt=1. Sun Feb 20 21:28:54 2022
bootstrapping rt=201. Sun Feb 20 21:29:13 2022
bootstrapping rt=401. Sun Feb 20 21:29:32 2022
bootstrapping rt=601. Sun Feb 20 21:29:51 2022
bootstrapping rt=801. Sun Feb 20 21:30:10 2022
Now summarizing method=CbMPDiCbraya i=1 j=4. Sun Feb 20 21:30:29 2022
bootstrapping rt=1. Sun Feb 20 21:30:29 2022
bootstrapping rt=201. Sun Feb 20 21:30:48 2022
bootstrapping rt=401. Sun Feb 20 21:31:07 2022
bootstrapping rt=601. Sun Feb 20 21:31:25 2022
bootstrapping rt=801. Sun Feb 20 21:31:44 2022
Now summarizing method=CbMPDiCbraya i=1 j=5. Sun Feb 20 21:32:03 2022
bootstrapping rt=1. Sun Feb 20 21:32:04 2022
bootstrapping rt=201. Sun Feb 20 21:32:22 2022
bootstrapping rt=401. Sun Feb 20 21:32:42 2022
bootstrapping rt=601. Sun Feb 20 21:33:01 2022
bootstrapping rt=801. Sun Feb 20 21:33:19 2022
Now summarizing method=CbMPDiCbraya i=1 j=6. Sun Feb 20 21:33:38 2022
bootstrapping rt=1. Sun Feb 20 21:33:39 2022
bootstrapping rt=201. Sun Feb 20 21:33:58 2022
bootstrapping rt=401. Sun Feb 20 21:34:16 2022
bootstrapping rt=601. Sun Feb 20 21:34:35 2022
bootstrapping rt=801. Sun Feb 20 21:34:54 2022
Now summarizing method=CbMPDiCbraya i=1 j=7. Sun Feb 20 21:35:12 2022
bootstrapping rt=1. Sun Feb 20 21:35:13 2022
bootstrapping rt=201. Sun Feb 20 21:35:32 2022
bootstrapping rt=401. Sun Feb 20 21:35:51 2022
bootstrapping rt=601. Sun Feb 20 21:36:10 2022
bootstrapping rt=801. Sun Feb 20 21:36:29 2022
Now summarizing method=CbMPDiCbraya i=1 j=8. Sun Feb 20 21:36:48 2022
Error in oldname[j, 1:2] : subscript out of bounds
Calls: icamp.bins -> <Anonymous> -> t -> sapply -> lapply -> FUN
Execution halted

**Comment by DaliangNing** [comment]
If you can send me (ningdaliang@ou.edu) your input files and R code, I will try to fix the problem.

**Comment by INFINITY1993** [comment]
Dear Ning, Thanks so much for your reply. After browsing the source code of icamp.bins, I think I have solved the problem. That is, in the treat file, each group should comprise of at least two samples.

          for (i in 1:ncol(treat)) {
                trti.lev = unique(as.vector(treat[, i]))
                for (j in 1:length(trti.lev)) {
                  if (!silent) 
                    message("Now summarizing method=", 
                      mij.name, " i=", i, " j=", 
                      j, ". ", date())
                  **sampij** = rownames(treat)[which(treat[, i] == 
                    trti.lev[j])]
                  bwij = bin.weight[which((bin.weight[, 1] %in% 
                    sampij) & (bin.weight[, 2] %in% sampij)), 
                    , drop = FALSE]

**Comment by INFINITY1993** [comment]
Dear, Ning. Another question confused me is how should I select n.min for the icamp.big step. I have summarized the phylosignal report as attached file shows. is "n.min=48" the best choice?
[PhyloSignalSummary.xlsx](https://github.com/DaliangNing/iCAMP1/files/8122451/PhyloSignalSummary.xlsx)

**Comment by DaliangNing** [comment]
Yes, the best choice is 48 according to your results. Then, you may set bin.size.limit = 48 in icamp.big.

**Comment by INFINITY1993** [comment]
OK! Thank you so much!
